### ISSUE TITLE
New DataSource: aws_dx_gateway

### ISSUE THREAD

**Issue by sleterrier** [terraform-providers/terraform-provider-aws#4983]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Following the addition of aws_dx_gateway and aws_dx_gateway_assocation resources, it would very useful to have access to an 'aws_dx_gateway' DataSource to allow for dynamic and variable based configurations. Particularly relevant in a multi regions/accounts scenario.  See example below.

### New or Affected Resource(s)

* aws_dx_gateway
* aws_dx_gateway_assocation

### Potential Terraform Configuration

```hcl
# Global  role
resource "aws_dx_gateway" "test" {
  name            = "${var.dxgw_name}"
  amazon_side_asn = "${ ... }"
}

# Regional role
data "aws_dx_gateway" "test" {
  count    = "${var.is_aws_dx_enable ? 1 : 0}"
  name     = "${var.dxgw_name}"
}

resource "aws_vpn_gateway" "test" {
  vpc_id = "${aws_vpc.test.id}"
}

resource "aws_dx_gateway_association" "test" {
  count          = "${var.is_aws_dx_enable ? 1 : 0}"
  dx_gateway_id  = "${data.aws_dx_gateway.test.id}"
  vpn_gateway_id = "${aws_vpn_gateway.test.id}"
}

```

### References

* https://github.com/terraform-providers/terraform-provider-aws/pull/2861
* https://github.com/terraform-providers/terraform-provider-aws/pull/4896

**Comment by bflad** [comment]
I can submit a pull request for this today or tomorrow. 👍

**Comment by bflad** [comment]
Pull request submitted: #4988

**Comment by bflad** [comment]
A new `aws_dx_gateway` data source has been merged and will release with version 1.25.0 of the AWS provider, likely tomorrow.

**Comment by bflad** [comment]
This has been released in [version 1.25.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1250-june-27-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
